### PR TITLE
v1.3: Vector search

### LINKED ISSUE #2477: v1.3: New experimental feature: Vector search
v1.3 introduces a new experimental feature, vector search.

---

> ### Vector search fundamentals
> _Vectors_ are general mathematical structures comprised of size and direction. They are used across programming fields, including 3D graphics and Large Language Models (LLMs).
> 
> _Embedding models_ are created by passing significant amounts of labeled data to a neural network.
> 
> _Vector embeddings_ are data structures storing semantic information about a dataset and its terms. Embeddings are generated by parsing raw data through an embedding model. Vector embeddings are the building blocks of many machine learning applications.

---

In this iteration, Meilisearch works as a vector store: users can index and search through documents using vector data.

Meilisearch does not generate nor calculate vector data. When adding documents, users must generate embeddings through an external tool. User must also use an external tool to compute vectors for a query.

Currently, vector search and Meilisearch's default keyword search are mutually exclusive. If a query uses the `vector` search parameter, Meilisearch ignores the `q` parameter.

Engineering has not yet decided whether this feature will be experimental—and thus gated behind an `--experimental` CLI flag—or stable and available by default.

## Tasks
- [x] Create experimental page: `/learn/experimental/vector-search.mdx`
- [x] Add new page to `sidebar-learn.json`

## References
- engine issue: https://github.com/meilisearch/meilisearch/issues/3798
- product feature discussion: https://github.com/meilisearch/product/discussions/621
- CTO Meilisearch vector search explainer: https://github.com/meilisearch/meilisearch/issues/3838
- PRD (private): https://www.notion.so/meilisearch/Vector-Search-cd5b8d6ce94944618914ab8d958aceb0
- vector search 101 (private): https://www.notion.so/meilisearch/Vector-Search-Starter-Guide-22430c88ec33488d853b7340d106f831
- SME: @Kerollmops 
- external resources
  - vector databases: https://www.pinecone.io/learn/vector-database/
  - vector embeddings and vectors: https://www.pinecone.io/learn/vector-embeddings-for-developers/

### PR DESCRIPTION
**Pull request by guimachiavelli**
Closes #2477

### CODE REVIEW

**Review by Kerollmops — APPROVED** [review]
I like what you wrote! Thank you 😎

File: config/sidebar-learn.json [hunk 1]
```diff
@@ -394,10 +394,15 @@
 							"slug": "reduce-indexing-memory-usage"
 					},
 					{
-						"source": "learn/experimental/ranking_rule_score_details.mdx",
-						"label": "Ranking score details",
-						"slug": "ranking_rule_score_details"
-				  }
+						  "source": "learn/experimental/ranking_rule_score_details.mdx",
+						  "label": "Ranking score details",
```
Comment (dichotommy) [inline config/sidebar-learn.json:398]: Not related to vector search but I think "Detailed ranking scores" is a clearer title for this page.

File: learn/experimental/vector_search.mdx [hunk 2]
```diff
@@ -0,0 +1,14 @@
+
+# Vector search
+
+Vector search is an experimental technology that uses Large Language Models to retrieve search results. This feature is useful when you are interested in results based on the meaning and context of a query, such as suggesting related products in webshop or articles in an online magazine.
```
Comment (dichotommy) [inline learn/experimental/vector_search.mdx:4]:
```suggestion
Vector search is an experimental technology that uses machine learning to retrieve search results. This feature is useful when you are interested in results based on the meaning and context of a query, such as suggesting related products in webshop or articles in an online magazine.
```

My understanding is that large language models need vector search to function, but vector search can be useful even without a large language model. Take or leave this suggestion as you like.

File: learn/experimental/vector_search.mdx [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+[Visit the GitHub discussion to learn how to use vector search](https://github.com/meilisearch/product/discussions/621).
```
Comment (dichotommy) [inline learn/experimental/vector_search.mdx:6]:
```suggestion
[Visit the GitHub discussion to learn how to get started with vector search in Meilisearch](https://github.com/meilisearch/product/discussions/621).
```

File: learn/experimental/vector_search.mdx [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+
+Are you using this feature? Meilisearch wants to hear from you! Tell us about your experience in the [GitHub discussion](https://github.com/meilisearch/product/discussions/621). All feedback is useful and helps make Meilisearch better and easier to use.
```
Comment (dichotommy) [inline learn/experimental/vector_search.mdx:8]:
```suggestion
Are you using this feature? We want to hear from you! Tell us about your experience in the [GitHub discussion](https://github.com/meilisearch/product/discussions/621). All feedback is useful and helps make Meilisearch better and easier to use.
```

Idk why but I always find it creepy when text personifies companies in this way. "Meilisearch wants to hear from you!" 😈

**Review by dichotommy — COMMENTED** [review]
No really useful feedback here, except for perhaps replacing "large language models" with "machine learning". Otherwise it lgtm

**Pull request merged**